docs/src/reference/the-traversal.asciidoc

```diff
@@ -2302,6 +2302,281 @@ link:++https://tinkerpop.apache.org/javadocs/x.y.z/core/org/apache/tinkerpop/gre
+existing vertex to occur, all values in the map must exist on a vertex; otherwise, a new
+vertex will be created. The examples that follow show how `mergeV` can be used to add some
+dogs to the graph.
+
+[gremlin-groovy]
+----
+g.mergeV(['name': 'Brandy']) <1>
+g.V().has('name','Brandy')
+g.mergeV([(T.label):'Dog','name':'Scamp', 'age':12]) <2>
+g.V().hasLabel('Dog').valueMap()
+g.mergeV([(T.id):300, (T.label):'Dog', 'name':'Toby', 'age':10]) <3>
+g.V().hasLabel('Dog').valueMap().with(WithOptions.tokens)
+----
+
+<1> Create a vertex for Brandy as no other matching ones exist yet.
+<2> Create a vertex for Scamp and also add a Dog label his age.
+<3> Create a vertex for Toby with an ID of 300.
+
+If a vertex already exists that matches the map passed to `mergeV`, the existing
+vertex will be returned, otherwise a new one will  be created. In this way, `mergeV`
+provides "get or create" semantics.
+
+[gremlin-groovy,existing]
+----
+g.mergeV(['name': 'Brandy']) <1>
+----
+
+<1> A vertex for Brandy already exists so return it. A new one is not created.
+
+It's important to note that every key/value pair passed to `mergeV` must already exist on
+one or more vertices for there to be a match. If a match is found, the vertex, or
+vertices, representing that match will be returned. If a vertex representing a dog called
+Brandy already exists, but it does not have an `age` property, the `mergeV` below will not
+find a match and a new vertex will be created.
+
+[gremlin-groovy]
+----
+g.addV('Dog').property('name','Brandy') <1>
+g.mergeV([(T.label):'Dog','name':'Brandy','age':13]) <2>
+----
+
+<1> Create a vertex for Brandy with no age property.
+<2> A new vertex is created as there is no exact match to any existing vertices.
+
+A common scenario is to search for a vertex with a known ID and if it exists return that
+vertex. If it does not exist, create it. As we have seen, one way to do this is to pass
+the ID and all properties directly to `mergeV`. Another is to use `Merge.onCreate`. Note
+that the map specified for `Match.onCreate` must include the ID if you want to create the
+vertex with the ID originally searched for. The values provided to the `mergeV` map are
+not combined with the map provided to `Merge.onCreate`.
+
+[gremlin-groovy]
+----
+g.mergeV([(T.id):300]).
+  option(Merge.onCreate,[(T.id):300, (T.label):'Dog', 'name':'Toby', 'age':10])
+----
+
+To take specific action when the vertex already exists, `Merge.onMatch` can be used. The
+second parameter to the `option` step can be either a map whose values are used to update
+the vertex or another Gremlin traversal that generates a map.
+
+
+NOTE: If `mergeV` is given an empty map; such as `mergeV([:])`, it will match, and
+return, every vertex in the graph. This is the same behavior seen with `V([])`.
+
+[gremlin-groovy]
+----
+g.mergeV([(T.id):300]).
+  option(Merge.onCreate,[(T.id):300, (T.label):'Dog', 'name':'Toby', 'age':10]). <1>
+  option(Merge.onMatch,['age':11]) <2>
+    
```

Comment:
you might want to remove these linefeeds - i'm pretty sure that they will render as empty `gremlin>` lines when the docs get generated.